swift/extractor/main.cpp

```diff
@@ -1,27 +1,32 @@
 #include <fstream>
 #include <iomanip>
 #include <stdlib.h>
+#include <unordered_set>
+#include <vector>
+#include <string>
+#include <iostream>
 
 #include <swift/Basic/LLVMInitialize.h>
 #include <swift/FrontendTool/FrontendTool.h>
 
 #include "SwiftExtractor.h"
+#include "SwiftOutputRewrite.h"
 
 using namespace std::string_literals;
 
 // This is part of the swiftFrontendTool interface, we hook into the
 // compilation pipeline and extract files after the Swift frontend performed
-// semantic analysys
+// semantic analysis
 class Observer : public swift::FrontendObserver {
  public:
   explicit Observer(const codeql::SwiftExtractorConfiguration& config) : config{config} {}
 
   void parsedArgs(swift::CompilerInvocation& invocation) override {
-    // Original compiler and the extractor-compiler get into conflicts when
-    // both produce the same output files.
-    // TODO: change the final artifact destinations instead of disabling
-    // the artifact generation completely?
-    invocation.getFrontendOptions().RequestedAction = swift::FrontendOptions::ActionType::Typecheck;
+    auto& overlays = invocation.getSearchPathOptions().VFSOverlayFiles;
```

Comment:
It can be empty or it may contain something in it already. As far as I can tell, this code adds new file system "views" on top of each other, so there is a risk that we override something here.

This also needs more testing in a real-world setting, but so far I didn't see this option used with several rather big projects.